homeassistant/components/amberelectric/sensor.py

```diff
@@ -0,0 +1,591 @@
+        if self._channel_type == ChannelType.CONTROLLED_LOAD:
+            return "mdi:clock-outline"
+        return "mdi:transmission-tower"
+
+    @property
+    def unit_of_measurement(self):
+        """Return the sensors unit of measurement."""
+        return "¢/kWh"
+
+    @property
+    def native_value(self) -> str | None:
+        """Return the current price in c/kWh."""
+        forecasts = self._data_service.forecasts.get(self._channel_type)
+        if forecasts and len(forecasts) > 0:
+            if self._channel_type == ChannelType.FEED_IN:
+                return round(forecasts[0].per_kwh, 0) * -1
+            return round(forecasts[0].per_kwh, 0)
+        return None
+
+    @property
+    def device_state_attributes(self) -> Mapping[str, Any] | None:
+        """Return additional pieces of information about the forecast."""
+        forecasts = self._data_service.forecasts.get(self._channel_type)
+        data: dict[str, Any] = {}
+        data["forecasts"] = []
+        data["channel_type"] = self._channel_type.value
+
+        if forecasts is not None:
+            for meta in forecasts:
+                datum = {}
+                datum["duration"] = meta.duration
+                datum["date"] = meta.date.isoformat()
+                datum["nem_date"] = meta.nem_time.isoformat()
+                datum["per_kwh"] = round(meta.per_kwh)
+                if self._channel_type == ChannelType.FEED_IN:
+                    datum["per_kwh"] = datum["per_kwh"] * -1
+                datum["spot_per_kwh"] = round(meta.spot_per_kwh)
+                datum["start_time"] = meta.start_time.isoformat()
+                datum["end_time"] = meta.end_time.isoformat()
+                datum["renewables"] = round(meta.renewables)
+                datum["spike_status"] = meta.spike_status.value
+
+                if meta.range is not None:
+                    datum["range_min"] = meta.range.min
+                    datum["range_max"] = meta.range.max
+
+                data["forecasts"].append(datum)
+
+        data[ATTR_ATTRIBUTION] = ATTRIBUTION
+        return data
+
+
+class AmberPriceSpikeSensor(CoordinatorEntity, SensorEntity):
+    """Amber Price Spike Sensor."""
+
+    def __init__(
+        self,
+        platform_name: str,
+        site_id: str,
+        data_service: AmberDataService,
+        coordinator: DataUpdateCoordinator,
+    ) -> None:
+        """Initialize the Sensor."""
+        super().__init__(coordinator)
+        self._platform_name = platform_name
+        self._site_id = site_id
+        self._data_service = data_service
+
+    @property
+    def name(self) -> str | None:
+        """Return the friendly name of the sensor."""
+        return self._platform_name + " - Price Spike"
+
+    @property
+    def unique_id(self) -> str | None:
+        """Return a unique id for each sensors."""
+        return slugify(self._site_id + " Price Spike")
+
+    @property
+    def native_value(self) -> bool:
```

Comment:
native value of a sensor is not a boolean. You are thinking of a binary sensor. 

Initial PR should only have 1 platform, so remove this one completely and open a 2nd PR later that adds this as a binary sensor